hal/halx86/acpi/halacpi.c

```diff
@@ -871,35 +871,86 @@ HalpSetupAcpiPhase0(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
+        // v1.0+: Revision is major version.
+        // v5.1+: minor_revision is minor version.
+        // v6.4+: errata bits are errata version.
+        switch (Fadt->Header.Revision)
+        {
+            case 0: // Should not happen.
+                AcpiVersion = "Unknown_0";
+                break;
+            case 1:
+                AcpiVersion = "1.0-1.0b";
+                break;
+            case 2: // Should not happen.
+                AcpiVersion = "Unknown_2";
+                break;
+            case 3:
+                AcpiVersion = "1.5-2.0_C";
+                break;
+            case 4:
+                AcpiVersion = "3.0-4.0_A";
+                break;
+            case 5:
+                if (Fadt->minor_revision == 0)
+                    AcpiVersion = "5.0-5.0_B";
+                else if (Fadt->minor_revision == 1)
+                    AcpiVersion = "5.1-5.1_B";
+                break;
+            case 6:
+                if (Fadt->minor_revision == 0)
+                    AcpiVersion = "6.0-6.0_A";
+                else if (Fadt->minor_revision == 1)
+                    AcpiVersion = "6.1-6.1_A";
+                else if (Fadt->minor_revision == 2)
+                    AcpiVersion = "6.2-6.2_B";
+                else if (Fadt->minor_revision == 3)
+                    AcpiVersion = "6.3-6.3_A";
+                else if ((Fadt->minor_revision & 0x0F) == 0x04)
+                {
+                    if ((Fadt->minor_revision & 0xF0) == 0x00)
+                        AcpiVersion = "6.4";
+                    else if ((Fadt->minor_revision & 0xF0) == 0x10)
+                        AcpiVersion = "6.4_A";
+                }
+                else if (Fadt->minor_revision == 5) // v6.5_A too is documented as errata=0.
+                    AcpiVersion = "6.5-6.6";
+                break;
+        }
 
-        /* Loop cached tables */
-        ListHead = &HalpAcpiTableCacheList;
-        NextEntry = ListHead->Flink;
-        while (NextEntry != ListHead)
+        /* Print the ACPI version */
+        DPRINT1("ACPI v");
+        if (AcpiVersion == NULL)
+        {
+            // Unknown past values, or newer than v6.6 (documented as 6.5).
+            DbgPrint("Unknown_%u_%u", Fadt->Header.Revision, Fadt->minor_revision);
+        }
+        else
         {
-            /* Get the table */
-            CachedTable = CONTAINING_RECORD(NextEntry, ACPI_CACHED_TABLE, Links);
+            DbgPrint("%s", AcpiVersion);
+        }
+        DbgPrint(" detected. Tables:");
 
-            /* Compare signatures */
-            if ((CachedTable->Header.Signature == RSDT_SIGNATURE) ||
-                (CachedTable->Header.Signature == XSDT_SIGNATURE))
-            {
-                DPRINT1("ACPI %u.0 Detected. Tables:", CachedTable->Header.Revision + 1);
-            }
+        /* List cached tables */
+        for (NextEntry = ListHead->Flink; NextEntry != ListHead; NextEntry = NextEntry->Flink)
```

Comment:
I wonder whether you could directly use `HalpAcpiTableCacheList` instead of the extra `ListHead` variable.

Reply:
Sure. Done.